### PR TITLE
add linear ODE solvers

### PR DESCRIPTION
**Pull request by Phiro333**
* add linear ImplicitEuler and MEBDF2 solvers
* add tests

### CODE REVIEW

**Comment by natschil** [comment]
Travis failing is probably an issue with the .travis.yml file not having DiffEqOperators installed for the documentation build step.

**Comment by dkarrasch** [comment]
Yes, but I wonder why the heck we need to load that many packages manually. We should try to understand that better and reduce that to a minimum at some point, i.e., in a different PR.

@Phiro333 For the time being, could you please replace

https://github.com/CoherentStructures/CoherentStructures.jl/blob/4327364fe8d8577cd1acf888fff85327e42c385d/.travis.yml#L32

by
```julia
- julia --color=yes --project=docs/ -e 'using Pkg; Pkg.activate(); Pkg.add("Documenter"); Pkg.add("Literate"); Pkg.add("AxisArrays"); Pkg.add("BenchmarkTools"); Pkg.add("Interpolations"); Pkg.add("Distances"); Pkg.add("JLD2"); Pkg.add("OrdinaryDiffEq"); Pkg.add("DiffEqOperators"); Pkg.add("Clustering"); Pkg.add("Arpack"); Pkg.add("Plots"); Pkg.add("StaticArrays"); Pkg.add("LinearMaps"); Pkg.add("Tensors"); Pkg.add(PackageSpec(url = "https://github.com/KristofferC/JuAFEM.jl.git")); Pkg.clone(pwd()); Pkg.build("Plots"); Pkg.resolve(); Pkg.instantiate(); using Plots'
```
?

File: Project.toml [hunk 1]
```diff
@@ -26,12 +27,13 @@ Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
 SymEngine = "123dc426-2d89-5057-bbad-38513e3affd8"
 Tensors = "48a634ad-e948-5137-8d70-aa71f2a747f4"
 VoronoiDelaunay = "72f80fcb-8c52-57d9-aff0-40c1a3526986"
-ProgressMeter = "92933f4c-e287-5a05-a399-4b506db050ca"
 
 [extras]
 BenchmarkTools = "6e4b80f9-dd63-53aa-95a3-0cdb28fa8baf"
+DiffEqDevTools = "f3b72e0c-5b89-59e1-b016-84e28bfd966d"
+DiffEqOperators = "9fdde737-9c7f-55bf-ade8-46b3f136cc48"
```
Comment (dkarrasch) [inline Project.toml]: Sorry, I mixed it up. `DiffEqOperators` is `import`ed in the main package, so this line should be moved to the normal dependencies above. I assume that we don't have to load it in the travis script, so the last change can be undone.

Comment (dkarrasch) [reply Project.toml]: Done.

**Pull request merged**